src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/VisitResourceIntTest.kt

```diff
@@ -0,0 +1,114 @@
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.transaction.PlatformTransactionManager
+import org.springframework.transaction.support.TransactionTemplate
+import org.springframework.web.reactive.function.BodyInserters
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateVisitRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateVisitResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderVisitBalanceAdjustmentRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.VisitRepository
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+private const val offenderBookingId = -10L
+private const val offenderNo = "A1234AJ"
+private const val prisonId = "BXI"
+private val createVisitRequest = CreateVisitRequest(
+  visitType = "SCON",
+  offenderNo = offenderNo,
+  startTime = LocalDateTime.of(2021, 11, 4, 12, 5),
+  endTime = LocalTime.of(13, 4),
+  prisonId = prisonId,
+  visitorPersonIds = listOf(-7L, -8L, -9L),
+  decrementBalances = true,
+)
+
+class VisitResourceIntTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var visitRepository: VisitRepository
+
+  @Autowired
+  lateinit var offenderVisitBalanceAdjustmentRepository: OffenderVisitBalanceAdjustmentRepository
+
+  @Autowired
+  lateinit var transactionManager: PlatformTransactionManager
+
+  @DisplayName("Create")
+  @Nested
+  inner class CreateVisitRequest {
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/visits")
+        .body(BodyInserters.fromValue(createVisitRequest))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/visits")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(createVisitRequest))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `create visit forbidden with wrong role`() {
+      webTestClient.post().uri("/visits")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .body(BodyInserters.fromValue(createVisitRequest))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `create visit with offender not found`() {
+      webTestClient.post().uri("/visits")
+        .headers(setAuthorisation(roles = listOf("ROLE_UPDATE_NOMIS")))
+        .body(BodyInserters.fromValue(createVisitRequest.copy(offenderNo = "Z9999ZZ")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `create visit success`() {
+      val response = webTestClient.post().uri("/visits")
+        .headers(setAuthorisation(roles = listOf("ROLE_UPDATE_NOMIS")))
+        .body(BodyInserters.fromValue(createVisitRequest))
```

Comment:
I think this would benefit if this was a inline String with the JSON rather than using CreateVisitRequest, given clients wouldn't use the CreateVisitRequest it helps to see the exact JSON being sent rather than relying on Jackson to convert an object to JSON.

Reply:
Done